x-pack/plugins/observability_onboarding/public/components/app/custom_logs/wizard/index.tsx

```diff
@@ -44,15 +45,22 @@ const initialState: WizardState = {
   onboardingId: '',
 };
 
-const { Provider, Step, useWizard } = createWizardContext({
+export type CustomLogsSteps =
+  | 'selectLogs'
+  | 'configureLogs'
+  | 'installElasticAgent'
+  | 'inspect';
+
+const steps: Record<CustomLogsSteps, ComponentType<{}>> = {
+  selectLogs: SelectLogs,
+  configureLogs: ConfigureLogs,
+  installElasticAgent: InstallElasticAgent,
+  inspect: Inspect,
+};
+
+const { Provider, useWizard } = createWizardContext({
```

Comment:
Is it possible to have `createWizardContext` also return  `Routes`, which can be generated based on the step keys. Perhaps passing in `basePath: '/customLogs'` as a param. Then `Routes` can be rendered as a component, instead of having to define each route and map thru them. I think we could avoid some boilerplate for custom logs and the next onboarding wizard if these routes were generated automatically.

Reply:
I really like this suggestion, let me check what I can do in that route 😆

Reply:
Addressed in [27f765c](https://github.com/elastic/kibana/pull/161230/commits/27f765c86c5254d17f8804d6fb6fb679afd6d00f).